ApiIndex/Program.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using Google.Protobuf.Reflection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Google.Cloud.Tools.ApiIndex
+{
+    internal static class Program
+    {
+        private static int Main(string[] args)
+        {
+            if (args.Length != 3)
+            {
+                Console.WriteLine("Expected arguments: <descriptor-set file> <googleapis directory> <output directory>");
+                return 1;
+            }
+
+            using var descriptorSetInput = File.OpenRead(args[0]);
+            FileDescriptorSet descriptorSet = FileDescriptorSet.Parser.ParseFrom(descriptorSetInput);
+
+            var apiModels = ApiModel.Load(descriptorSet, args[1]);
+
+            WriteV1Index(args[2], apiModels);
+
+            return 0;
+        }
+
+        private static void WriteV1Index(string outputDirectory, IEnumerable<ApiModel> apiModels)
```

Comment:
Maybe this whole project needs to be ApiIndex.V1, and if at some point several index versions coexist, the common code can be extracted to a library or something?

Reply:
I'd definitely prefer to anticipate multiple versions from the start, separating out "everything we can find out" from "the representation in the schema at the moment".